### PR TITLE
MINOR: WakeupTrigger cleanup

### PR DESCRIPTION
**Pull request by kirktrue**
Added comments, made package-visible, and removed inner interface and classes.

### CODE REVIEW

**Comment by kirktrue** [comment]
@philipnee Would you tag this with `ctr` and review?

This is a low priority clean up.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java [hunk 1]
```diff
@@ -24,85 +25,133 @@
...
+     *         internal indicator that the <em>next</em> attempt to start a long-running task (via a call to
+     *         {@link #setActiveTask(CompletableFuture)}) will fail with a {@link WakeupException}.
+     *     </li>
+     *     <li>
+     *         If there is an <em>active</em> task (i.e. there was a previous call to
+     *         {@link #setActiveTask(CompletableFuture)} for a long-running task), fail it via
+     *         {@link CompletableFuture#completeExceptionally(Throwable)} and then clear the <em>active</em> task.
+     *     </li>
+     *     <li>
+     *         If there is already an pending wakeup from a previous call to {@link Consumer#wakeup()}, do nothing.
```
Comment (philipnee) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java:54]: "a pending"

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java [hunk 2]
```diff
@@ -24,85 +25,133 @@
...
+     * <ul>
+     *     <li>
+     *         If there is no <em>active</em> task from a previous call to {@code setActiveTask}, we set an
+     *         internal indicator that the <em>next</em> attempt to start a long-running task (via a call to
+     *         {@link #setActiveTask(CompletableFuture)}) will fail with a {@link WakeupException}.
+     *     </li>
+     *     <li>
+     *         If there is an <em>active</em> task (i.e. there was a previous call to
+     *         {@link #setActiveTask(CompletableFuture)} for a long-running task), fail it via
+     *         {@link CompletableFuture#completeExceptionally(Throwable)} and then clear the <em>active</em> task.
```
Comment (philipnee) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java:51]: should we mention about WakeupException - something like active task compelted exceptionally via WakeupException

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java [hunk 3]
```diff
@@ -24,85 +25,133 @@
...
+     *
+     * <ul>
+     *     <li>
+     *         If there is no <em>active</em> task from a previous call to {@code setActiveTask} <em>and</em> no
+     *         previous calls to {@link #wakeup()}, set the given {@link CompletableFuture} as the
+     *         <em>active</em> task.
+     *     </li>
+     *     <li>
+     *         If there was a previous call to {@link #wakeup()}, the given {@link CompletableFuture task} will fail
+     *         via {@link CompletableFuture#completeExceptionally(Throwable)} and the <em>active</em> task will be
```
Comment (philipnee) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/WakeupTrigger.java:99]: similarly - should we mention WakeupException

**Comment by philipnee** [comment]
Hi @kirktrue Thanks for taking time to clean up the task. I left some comments there.

**Pull request closed without merging**